### PR TITLE
Add Base64

### LINKED ISSUE #152: Support `base64`
The XML-RPC protocol supports sending binary files with the `base64` datatype
https://en.wikipedia.org/wiki/XML-RPC#Data_types
Any chance of adding this to `ocaml-rpc`? :)

### PR DESCRIPTION
**Pull request by CGenie**
Closes #152

### CODE REVIEW

File: src/lib/rpc.ml [hunk 1]
```diff
@@ -167,6 +169,8 @@ let rec to_string t =
   | Dict ts ->
       sprintf "{%s}"
         (map_strings ";" (fun (s, t) -> sprintf "%s:%s" s (to_string t)) ts)
+  | Base64 s ->
+      sprintf "S(%s)" s
```
Comment (mseri) [inline src/lib/rpc.ml]: Should let’s use `B64` Instead of `S` Here to distinguish the types

File: src/lib/jsonrpc.ml [hunk 2]
```diff
@@ -43,6 +43,7 @@ let rec rpc_to_json t =
   | Float r -> `Float r
   | String s -> `String s
   | DateTime d -> `String d
+  | Base64 b -> `String b
```
Comment (mseri) [inline src/lib/jsonrpc.ml]: It’s really annoying that in json we don’t have a way to infer the type. Please add a comment on `json_to_rpc` as we have for the DateTime

**Comment by mseri** [comment]
Thank you very much!

File: rpc.opam [hunk 3]
```diff
@@ -14,6 +14,7 @@ depends: [
   "rpclib-lwt" {=version}
   "ppx_deriving_rpc" {=version}
   "async" {>= "v0.9.0"}
+  "base64" {>= "3.4.0"}
```
Comment (mseri) [inline rpc.opam]: This needs to be added to rpclib.opam

Comment (mseri) [reply rpc.opam]: rpc is a legacy virtual package

**Comment by mseri** [comment]
I think also he ppxs need to be updated

**Comment by CGenie** [comment]
@mseri Thank you for your comments, I updated the code. About ppxs, I'm not sure where to add this.
So far I just added this:
https://github.com/mirage/ocaml-rpc/pull/154/files#diff-36b922328d60097472c823ecbc21c056R15

File: ppx/common.ml [hunk 4]
```diff
@@ -11,7 +11,8 @@ let core_types loc =
     ; ("int64", [%expr Rpc.Types.(Basic Int64)])
     ; ("string", [%expr Rpc.Types.(Basic String)])
     ; ("float", [%expr Rpc.Types.(Basic Float)])
-    ; ("bool", [%expr Rpc.Types.(Basic Bool)]) ]
+    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
+    ; ("base64", [%expr Rpc.Types.(Basic Base64)]) ]
```
Comment (mseri) [inline ppx/common.ml]: Please revert this change. 
At this stage, base64 is not a core type. Like DateTime it needs to be handled by hand.

**Comment by mseri** [comment]
Just a small change and it's good to merge imo

File: ppx/common.ml (hunk 4 above)
Comment (CGenie) [reply ppx/common.ml]: OK done.

**Comment by mseri** [comment]
Thanks! I'll wait for Travis and merge

File: ppx/common.ml [hunk 5]
```diff
@@ -11,8 +11,7 @@ let core_types loc =
     ; ("int64", [%expr Rpc.Types.(Basic Int64)])
     ; ("string", [%expr Rpc.Types.(Basic String)])
     ; ("float", [%expr Rpc.Types.(Basic Float)])
-    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
-    ; ("base64", [%expr Rpc.Types.(Basic Base64)]) ]
+    ; ("bool", [%expr Rpc.Types.(Basic Bool)]) ]) ]
```
Comment (mseri) [inline ppx/common.ml]:
```suggestion
    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
    ]
```
